### ISSUE TITLE
Duplicated comand-line switches

### ISSUE THREAD

**Issue by msaaltink** [GaloisInc/saw-script#1363]
$ saw -?
    Usage: saw [OPTION...] [-I | file]
      -h, -?    --help                 Print this help message
      -V        --version              Show the version of the SAWScript interpreter
      -c path   --classpath=path       Colon-delimited list of paths
      -i path   --import-path=path     Colon-delimited list of paths
      -t        --detect-vacuity       Checks and warns the user about contradictory assumptions. (default: false)
      -t        --extra-type-checking  Perform extra type checking of intermediate values
      [...]

(output reformatted and truncated to fit here; see #405).  There are two options with short name "-t"!  Was that intentional?  Does specifying "-t" turn both checks on?

**Comment by robdockins** [comment]
The `detect-vacuity` option was added fairly recently, in efa72774e.  I suspect duplicating "-t" was unintentional.

**Comment by Ptival** [comment]
Yep, my bad!

**Comment by kquick** [comment]
It would be great if you found something in the parser that could help us avoid this in the future as you're fixing this, Val.

**Comment by Ptival** [comment]
I'm not sure the current API allows us much static checking, short of adding some fairly complex dependent type tricks when building the list of options.

It'd be fairly easy to add a loud dynamic check when executing `saw/Main`, that would maybe trip up the CI if placed at the appropriate location. Is this something we'd want?

**Comment by kquick** [comment]
I don't think we want to go to extraordinary lengths (e.g. complex dependent type tricks), but a dynamic check that was only executed during CI (e.g. maybe a test-only check of arguments) would be nice.

**Comment by sauclovian-g** [comment]
...three years later there are still two -t options...

**Comment by sauclovian-g** [comment]
I will fix this as part of #405, which is going to require tidying the way the options table is managed anyway.

**Comment by sauclovian-g** [comment]
On the plus side it looks like the getopt library we're using fails if you try to use an ambiguous argument, so all this time neither of the -t options actually works. Which means we can change one of them (presumably the newer one) without breaking anyone. What shall we use for --detect-vacuity? -x doesn't seem to be in use.

**Comment by sauclovian-g** [comment]
Yes, you can just set the short options to "". I agree, I'll do that.

(also now that I think about it, I wonder if it ought to be a command-line option or a setting in saw-script or what, but we can worry about that some other time)

**Comment by sauclovian-g** [comment]
The bug part's fixed now.

**Comment by sauclovian-g** [comment]
I think the table is sufficiently legible now that the risks of getting duplicates by accident are substantially reduced, so I'm inclined to close this without adding an explicit check.

however... I notice that the duplicate -t options were nonetheless on adjacent lines originally, so maybe this isn't enough and I should add a check too. @kquick what do you think?
